src/sv-pipeline/scripts/make_scramble_vcf.py

```diff
@@ -0,0 +1,511 @@
+            continue
+        row = {columns[i]: record[i] for i in range(len(record))}
+
+        row['sides'] = 1
+        _cast_numeric(row, FLOAT_COLUMNS, float)
+        _cast_numeric(row, INT_COLUMNS, int)
+        new_buffer = list()
+        found_match = False
+        for item in buffer:
+            if found_match:
+                new_buffer.append(item)
+            elif item['chrom'] != row['chrom'] \
+                    or abs(int(item['pos']) - int(row['pos'])) > cluster_distance:
+                item['svlen'] = _calculate_svlen_one_sided(item)
+                data.insert(0, item)
+            elif item['Clipped_Side'] != row['Clipped_Side'] \
+                    and item['Insertion_Direction'] == row['Insertion_Direction']:
+                row['pos'] = item['pos']
+                row['svlen'] = _calculate_svlen_two_sided(item, row)
+                data.append(row)
+                found_match = True
+            else:
+                new_buffer.insert(0, item)
+        if not found_match:
+            new_buffer.insert(0, row)
+        buffer = new_buffer
+    for item in buffer:
+        item['svlen'] = _calculate_svlen_one_sided(item)
+        data.append(item)
+    # Sort records in memory
+    data = sorted(data, key=lambda x: _record_sort_key(record=x, ref_sequences_index_dict=ref_sequences_index_dict))
+    return pd.DataFrame(data=data)
+
+
+def fails_indel_filter(record, samfile, mei_trees, indel_window, min_reads, min_ins_size, max_ins_size,
+                       min_del_size, max_del_size):
+    """
+    Checks if a given record lies within a reference MEI and is sufficiently close to a nearby indel,
+    based on local read alignments, which is a common error mode for Scramble.
+    Args:
+        record: VariantRecord
+            Variant in question
+        samefile: AlignmentFile
+            SAM/BAM/CRAM file opened with pysam
+        mei_trees: dict of IntervalTree
+            Reference MEI intervals
+        indel_window: int
+            Window in bp to look for indels around the insertion site
+        min_reads: int
+            Required number of reads containing an indel to filter
+        min_ins_size: int
+            Minimum insertion length to look for in the read CIGAR
+        max_ins_size: int
+            Maximum insertion length to look for in the read CIGAR
+        min_del_size: int
+            Minimum deletion length to look for in the read CIGAR
+        max_del_size: int
+            Maximum deletion length to look for in the read CIGAR
+    Returns:
+        True if the call should be filtered
+    """
+    def _get_num_indels(reads, min_reads, min_ins_size, max_ins_size, min_del_size, max_del_size):
+        ins_count = 0
+        del_count = 0
+        for read in reads:
+            if read.cigartuples is None:
+                continue
+            for tup in read.cigartuples:
+                if tup[0] == CIGAR_TUPLE_INS_INDEX and min_ins_size <= tup[1] <= max_ins_size:
```

Comment:
Good question, IIRC I found that indels within simple repeats were a big problem because they tended to be scattered in space. This was a enough of a problem that I had to group the count together over the full window. I'll add a comment with an explanation.